osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs

```diff
@@ -1,12 +1,96 @@
 ﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.OpenGL.Vertices;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Shaders;
+using osu.Framework.Graphics.Textures;
+using osu.Framework.Input;
+using osu.Framework.Input.Events;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.UI;
+using OpenTK;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModFlashlight : ModFlashlight
+    public class OsuModFlashlight : ModFlashlight, IApplicableToRulesetContainer<OsuHitObject>
     {
         public override double ScoreMultiplier => 1.12;
+
+        public void ApplyToRulesetContainer(RulesetContainer<OsuHitObject> rulesetContainer)
+        {
+            rulesetContainer.KeyBindingInputManager.Add(new Flashlight
+            {
+                RelativeSizeAxes = Axes.Both,
+            });
+        }
+
+        private class Flashlight : Drawable, IRequireHighFrequencyMousePosition
```

Comment:
Why not part of ModFlashlight? In that case you need just to set right settings for any mods(OsuFlashlight : Flashlight).

Reply:
If making that generic i’d call it `CircularFlashlightMask` or similar.

Reply:
What about a box/circle boolean uniform inside the shader?